src/sinks/kafka.rs

```diff
@@ -285,7 +286,7 @@ impl Sink<Event> for KafkaSink {
         let timestamp_ms = match &item {
             Event::Log(log) => log
                 .get(log_schema().timestamp_key())
-                .and_then(|v| v.as_timestamp()),
+                .map(|v| v.as_timestamp()),
```

Comment:
`as_timestamp` will panic if the field is not a timestamp. Are we absolutely sure that this will never happen here?

Reply:
Nope. This can happen. Just send this message to the sink to get a panic:

```
{"message": "noog", "timestamp": "noog"}
```